src/markov/markov_approx.jl

```diff
@@ -219,3 +223,567 @@ function estimate_mc_discrete{T}(X::Vector{T})
+    gaussianMoment = zeros(nMoments)
+    c = 1
+    for k=1:floor(TI, nMoments/2)
+        c = (2*k-1)*c
+        gaussianMoment[2*k] = c
+    end
+    # Compute standardized VAR(1) representation
+    # (zero mean and diagonal covariance matrix)
+    A, C, mu, Sigma = standardize_var(b, B, Psi, M)
+
+    # Construct 1-D grids
+    y1D = construct_1D_grid(Sigma, Nm, M, nSigmas, method)
+
+    # Construct all possible combinations of elements of the 1-D grids
+    D = allcomb3(y1D')'
+
+    # Construct finite-state Markov chain approximation
+    # conditional mean of the VAR process at each grid point
+    condMean = A*D
+    # probability transition matrix
+    P = ones(Nm^M, Nm^M)
+    # normalizing constant for maximum entropy computations
+    scalingFactor = y1D[:, end]
+    # used to store some intermediate calculations
+    temp = Array{Float64}(M, Nm)
+    # store optimized values of lambda (2 moments) to improve initial guesses
+    lambdaBar = zeros(2*M, Nm^M)
+    # small positive constant for numerical stability
+    kappa = 1e-8
+
+    for ii = 1:(Nm^M)
+
+        # Construct prior guesses for maximum entropy optimizations
+        q = construct_prior_guess(condMean[:, ii], Nm, y1D, method)
+
+        # Make sure all elements of the prior are stricly positive
+        q[q.<kappa] = kappa
+
+        for jj = 1:M
+            # Try to use intelligent initial guesses
+            if ii == 1
+                lambdaGuess = zeros(2)
+            else
+                lambdaGuess = lambdaBar[(jj-1)*2+1:jj*2, ii-1]
+            end
+
+            # Maximum entropy optimization
+            if nMoments == 1 # match only 1 moment
+                temp[jj, :], _, _ = discrete_approximation(y1D[jj, :],
+                    X -> (reshape(X, 1, Nm)-condMean[jj, ii])/scalingFactor[jj],
+                    [0.0], reshape(q[jj, :], 1, Nm), [0.0])
+            else # match 2 moments first
+                p, lambda, momentError = discrete_approximation(y1D[jj, :],
+                    X -> polynomial_moment(X, condMean[jj, ii], scalingFactor[jj], 2),
+                    [0; 1]./(scalingFactor[jj].^(1:2)),
+                    reshape(q[jj, :], 1, Nm), lambdaGuess)
+                if norm(momentError) > 1e-5 # if 2 moments fail, just match 1 moment
+                    warn("Failed to match first 2 moments. Just matching 1.")
+                    temp[jj, :], _, _ = discrete_approximation(y1D[jj, :],
+                        X -> (X-condMean[jj,ii])/scalingFactor[jj],
+                        0, reshape(q[jj, :], 1, Nm), 0)
+                    lambdaBar[(jj-1)*2+1:jj*2, ii] = zeros(2,1)
+                elseif nMoments == 2
+                    lambdaBar[(jj-1)*2+1:jj*2, ii] = lambda
+                    temp[jj, :] = p
+                else # solve maximum entropy problem sequentially from low order moments
+                    lambdaBar[(jj-1)*2+1:jj*2, ii] = lambda
+                    for mm = 4:2:nMoments
```

Comment:
It's invalid number: see [this test](https://github.com/QuantEcon/QuantEcon.jl/pull/163/files#diff-b49beb3db45370dfe66a3118785f6919R105)